macro-fullcalendar-api/src/main/java/org/xwiki/fullcalendar/script/FullCalendarScriptService.java

```diff
@@ -52,4 +54,18 @@ public String iCalToJSON(String iCalStringURL) throws Exception
     {
         return fullCalendarManager.iCalToJSON(iCalStringURL);
     }
+
+    /**
+     * Convert an iCal to a JSON.
+     *
+     * @param iCalStringURL the String representation of an iCal URL.
+     * @param startDate the beginning of the interval in which the returned events should be.
+     * @param endDate the end of the interval.
+     * @return the JSON representation of a calendar.
+     * @throws Exception in case of exceptions.
+     * @since 2.3
+     */
+    public String iCalToJSON(String iCalStringURL, Date startDate, Date endDate) throws Exception {
```

Comment:
actually, now that we have this API, the other API, without dates, looks absurd (it doesn't make any sense to have an "infinite" calendar), so maybe we could deprecate it. We could also remove it but I understand that _maybe_ someone uses it, so let's just deprecate it...

Reply:
The other option is to consider that there are 2 usecases:
* transforming the _actual data_ from the iCal in JSON - whichever that is. This wouldn't need dates as it would be just a transcription of data from ical to JSON
* producing the events from the iCal calendar and then represent them JSON. In this case dates make sense, and also recurrence expansion.

It seems to me that this is what you did: you left the old function doing what it was doing and you made this new function which does something slightly different from a semantic point of view, but with the same name and with parameters that could also apply to the existing function; date parameters make sense also for a filtering of a calendar where recurrent events are not expanded, so it needs to be probably more clear than this that these 2 functions are actually doing different things, we may need to clarify these function names / APIs.